pkg/config/environment_containers.go

```diff
@@ -148,15 +148,28 @@ func isCriSupported() bool {
 }
 
 func detectFargate(features FeatureMap) {
-	if IsECSFargate() {
+	isECSFargate := IsECSFargate()
+	if isECSFargate {
 		features[ECSFargate] = struct{}{}
+
 		return
 	}
 
-	if Datadog.GetBool("eks_fargate") {
+	isEKSFargate := Datadog.GetBool("eks_fargate")
+	if isEKSFargate {
 		features[EKSFargate] = struct{}{}
 		features[Kubernetes] = struct{}{}
 	}
+
+	if isECSFargate || isEKSFargate {
```

Comment:
Return statement on line 155 means we will not reach this line if isECSFargate is true